src/cheri_insts.sail

```diff
@@ -2013,8 +1998,8 @@ function clause execute AMOSwapCap(cd, cs2, rs1_cs1, aq, rl) = {
                   match wres {
                     MemValue(_) => {
                       let cr = if ptw_info.ptw_lc == PTW_LC_CLEAR
-                        then invalidateCap(v) /* strip the tag */
-                        else invalidateCapUnless(v, auth_val.permit_load_cap);
+                        then clearTag(v) /* strip the tag */
+                        else clearTagIf(v, not(auth_val.permit_load_cap));
```

Comment:
Nit: perhaps simpler to just flatten this to
`let cr = clearTagIf(v, (ptw_info.ptw_lc == PTW_LC_CLEAR) | not(auth_val.permit_load_cap));`?

Reply:
Yeah, I think I'd left it this way for symmetry with the cases that still needed the `if` to do reservation things, but you're right that this is clearer.